### DISCUSSION TITLE
Enabling pLDDT coloring in plugin (v3.0.0)

### DISCUSSION THREAD

**Discussion by dhrblav** [molstar/molstar#354] [Q&A] [answered]
Hi,
If I load an AlphaFold entry in a v3.0.0 PluginUIContext, the new <Set Coloring/Validation/pLDDT Confidence> option available in the Viewer is not available in the plugin.
So I suppose extra steps must be taken to enable pLDDT in plugins.
Can you please help me with this?
Thank you,
Andre

**Comment by dsehnal** [comment, marked as answer]
Hi, the default viewer defines an extra preset which loads this: https://github.com/molstar/molstar/blob/master/src/apps/viewer/app.ts#L465 You should be able to reuse this code. If you need more help, feel free to ask.

**Reply by arose** [reply]
@dhrblav also note that pLDDT preset and coloring works only for AlphaFold DB's mmCIF files not for their PDB files. If you have to use the PDB files you can try adjusting the "Uncertainty" color theme which shows data from the B-factor column (or RMSF from IHM models).

**Reply by dhrblav** [reply]
OK, thank you both.

**Reply by adsodemelk** [reply]
@dsehnal Would this work for Multimer structures? It seems the default pLDDT preset works only on monomer AlphaFold output

**Reply by arose** [reply]
Do you have an example for a multimer structure? If it is not an mmCIF file with ma_qa_metric categories it is not supported by the pLDDT preset.

**Reply by adsodemelk** [reply]
Hello @arose . This is an example of a multimer prediction [prediction.cif.gz](https://github.com/molstar/molstar/files/8661762/prediction.cif.gz) It has the ma_qa_metric categories and the image has a comparison with af2 monomer 
![Screenshot from 2022-05-10 09-44-11](https://user-images.githubusercontent.com/4891574/167635608-a96748b2-b919-41a3-b8f0-58bc1508a7e3.png)
You will notice that this file has 2 chains and only one gets coloured when loading it into the `PDBeMolstarPlugin` 
```
<script>

  var viewerInstance = new PDBeMolstarPlugin();

  var options = {
    customData: {
      url: 'https://alphafold.ebi.ac.uk/files/AF-P05067-F1-model_v1.cif',
      format: 'mmcif',
    },
    alphafoldView: true,
    bgColor: {r:255, g:255, b:255},
    hideControls: false,

  }

  var viewerContainer = document.getElementById('myViewer');
  viewerInstance.render(viewerContainer, options);

</script>
```

**Reply by arose** [reply]
I can't open the prediction.cif.gz example file

**Reply by adsodemelk** [reply]
@arose thanks for having a look. I did use `tar -czf prediction.cif prediction.cif.gz` to compress it as github does not accept cif files. Apologies for this.
You can download the original file from [here](https://storage.googleapis.com/deepchain-datasets-public/cif/prediction.cif)
Thanks

**Reply by arose** [reply]
Thanks, that file works on molstar.org/viewer. You will need to wait until PDBe updates their plugin or use Mol* directly.

**Reply by adsodemelk** [reply]
Thanks for having a look @arose !
